crates/ruff/src/rules/pycodestyle/rules/tab_indentation.rs

```diff
@@ -16,15 +16,48 @@ impl Violation for TabIndentation {
 }
 
 /// W191
-pub fn tab_indentation(lineno: usize, line: &str) -> Option<Diagnostic> {
+pub fn tab_indentation(lineno: usize, line: &str, string_ranges: &[Range]) -> Option<Diagnostic> {
     let indent = leading_space(line);
+    if let Some(tab_index) = indent.find('\t') {
+        // If the tab character is within a multi-line string, abort.
+        if let Ok(range_index) = string_ranges.binary_search_by(|range| {
+            let start = range.location.row();
+            let end = range.end_location.row();
+            if start > lineno {
+                std::cmp::Ordering::Greater
+            } else if end < lineno {
+                std::cmp::Ordering::Less
+            } else {
+                std::cmp::Ordering::Equal
+            }
```

Comment:
@evanrittenhouse - Does this look reasonable, to use the built-in `binary_search_by` here?

Reply:
Yep! Wasn't aware it existed, seems much cleaner